src/main/java/net/minecraftforge/fml/common/langsupport/IPropertyWrapper.java

```diff
@@ -0,0 +1,101 @@
+package net.minecraftforge.fml.common.langsupport;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+
+public interface IPropertyWrapper
+{
+    String getName();
+
+    Class<?> getType();
+
+    void set(Object value, boolean ignoreAccess, boolean ignoreFinal) throws ReflectiveOperationException;
+
+    Object get(boolean ignoreAccess) throws ReflectiveOperationException;
+
+    boolean isAnnotationPresent(Class<? extends Annotation> annotationClass);
+
+    class FieldWrapper implements IPropertyWrapper
+    {
+        private static Field modifiersField;
+        private static Object reflectionFactory;
+        private static Method newFieldAccessor;
+        private static Method fieldAccessorSet;
+        private Object instance;
+        private final Field field;
+        private boolean removedAccessRestrictions;
+        private boolean removedFinal;
+
+        public FieldWrapper(Field field, Object instance)
+        {
+            this.field = field;
+            this.instance = instance;
+        }
+
+        @Override
+        public String getName()
+        {
+            return field.getName();
+        }
+
+        @Override
+        public Class<?> getType()
+        {
+            return field.getType();
+        }
+
+        @Override
+        public void set(Object value, boolean ignoreAccess, boolean ignoreFinal) throws ReflectiveOperationException
+        {
+            if (ignoreAccess && !removedAccessRestrictions)
+            {
+                field.setAccessible(true);
+                removedAccessRestrictions = true;
+            }
+            if (ignoreFinal)
+            {
+                removeFinal();
+            }
+            if (reflectionFactory == null)
```

Comment:
Ah, that's good to know. It is required since I don't intend to change any semantics of the current usages of statics in Forge classes and `@ObjectHolder`-annotated classes require fields to be `public static final`, so the new system also has to support it. I personally don't agree with the finality restriction on these fields, but I don't have any authority on it xD